_extras/snippets_library/UCL_sge/14/module_avail.snip

```diff
@@ -1,49 +1,802 @@
+cuda/8.0.61-patch2/gnu-4.9.2     paraver/4.6.4.rc1/bindist
+cuda/9.0.176-patch4/gnu-4.9.2    perl/5.16.0
+cudnn/5.1/cuda-7.5               perl/5.22.0
+cudnn/5.1/cuda-8.0               perlbrew/0.73
+cudnn/6.0/cuda-7.5               pigz/2.4
+cudnn/6.0/cuda-8.0               pycuda/2017.1/python2
+cudnn/7.0.4/cuda-8.0             pycuda/2017.1/python3
+cudnn/7.1.4/cuda-9.0             pypy3/6.0.0/gnu-4.9.2
+cudnn/7.4.2.24/cuda-10.0         python/2.7.12
+cudnn/7.4.2.24/cuda-9.0          python/2.7.9
+cudnn/7.5.0.56/cuda-10.0         python/3.4.3
+cudnn/7.5.0.56/cuda-10.1         python/3.5.2
+cudnn/7.6.5.32/cuda-10.0         python/3.6.1/gnu-4.9.2
+cudnn/7.6.5.32/cuda-10.1         python/3.6.3(default)
+ddt/6.0.4                        python/3.7.0
+depot_tools/788d9e0d             python/3.7.2
+depot_tools/c03a9cf              python/3.7.4
+dimemas/5.3.3/bindist            python/3.8.0
+doxygen/1.8.14                   python/idp3/2019/3.6.8
+emacs/24.5                       python/miniconda3/4.5.11
+emacs/26.3                       qt/4.8.6/gnu-4.9.2
+extrae/3.5.2/intel-2017          qt/5.12.1/gnu-4.9.2
+f2c/2013-09-26/gnu-4.9.2         qt/5.4.2/gnu-4.9.2
+flex/2.5.39                      qwt/6.1.4/gnu-4.9.2
+git/2.10.2                       racket/6.8
+git/2.19.1                       rappture/20130903
+git/2.3.5                        ruby/2.2.2
+gperf/3.0.4/gnu-4.9.2            ruse/1.0.1
+guile/2.0.11/gnu-4.9.2           sbcl/1.3.19
+haskellplatform/2014.2.0.0       scons/2.3.4
+htop/1.0.3/gnu-4.9.2             strace/4.12
+java/1.8.0_45                    subversion/1.8.13
+java/1.8.0_92                    swig/3.0.5/gnu-4.9.2
+java/openjdk-11/11.0.1           swig/3.0.7/gnu-4.9.2
+java/openjdk-11/11.0.3u7/hotspot tcl/8.6.8
+java/openjdk-11/11.0.3u7/openj9  v8/3.15
+java/openjdk-8/8u212/hotspot     v8/5.6
+java/openjdk-8/8u212/openj9      valgrind/3.11.0/gnu-4.9.2
+julia/0.3.10                     xbae/4.60.4
+julia/0.4.0                      xorg-utils/X11R7.7
+julia/0.4.7
+
+--------------------- /shared/ucl/apps/modulefiles/bundles ---------------------
+beta-modules                  octave/recommended
+bioperl/recommended           personal-modules
+blic-modules                  physics-modules
+cancerit/20190218             pypy3/3.5-compat
+cancerit/recommended          python2/recommended
+chemistry-modules             python3/3.4
+climate-tools/recommended     python3/3.5
+deep_earth                    python3/3.6
+default-modules/2015          python3/3.7
+default-modules/2017          python3/3.8
+default-modules/2018(default) python3/recommended(default)
+default-modules-aristotle     r/new
+economics-modules             r/old
+farr/recommended              r/r-3.6.0_bc-3.9
+farr-modules                  r/recommended
+gmt/new                       rsd-modules
+gmt/old                       thermo-modules
+gmt/recommended               torch-deps
+naglib/mark27-intel-2019      workaround-modules
```

Comment:
I get the output of all that in columns, we probably should keep it as such. Also, it's better to truncate the output as this is not a constant as it will keep changing as new software is installed.

Reply:
Hi @dpshelio The only way I found of getting the output of `module avail` into a file was with the command `module avail >out.txt 2>error.txt`. Unfortunately that removed the nice column formatting.

Reply:
This shouldn't be an exhaustive list of all the modules, as it's only used to demonstrate how to run that command and what's the kind of output shown. Apply the change I suggest above and remove the rest of the file.